core/src/main/java/org/mule/runtime/core/construct/AbstractPipeline.java

```diff
@@ -424,6 +430,10 @@ protected boolean useBlockingCodePath() {
     return isTransactionActive() || processingStrategy.isSynchronous();
   }
 
+  public Map<String, EventContext> getSerializedEventContextCache() {
```

Comment:
i would refactor this map into an object that contains this map. See below comments.